src/NuGet.Core/NuGet.PackageManagement/Utility/XmlUtility.cs

```diff
@@ -9,117 +9,22 @@ namespace NuGet.ProjectManagement
 {
     public static class XmlUtility
     {
-        public static XDocument LoadSafe(string filePath)
-        {
-            var settings = CreateSafeSettings();
-            using (var reader = XmlReader.Create(filePath, settings))
-            {
-                return XDocument.Load(reader);
-            }
-        }
-
-        public static XDocument LoadSafe(Stream input)
-        {
-            var settings = CreateSafeSettings();
-            var reader = XmlReader.Create(input, settings);
-            return XDocument.Load(reader);
-        }
-
-        public static XDocument LoadSafe(Stream input, bool ignoreWhiteSpace)
-        {
-            var settings = CreateSafeSettings(ignoreWhiteSpace);
-            var reader = XmlReader.Create(input, settings);
-            return XDocument.Load(reader);
-        }
-
-        public static XDocument LoadSafe(Stream input, LoadOptions options)
-        {
-            var settings = CreateSafeSettings();
-            var reader = XmlReader.Create(input, settings);
-            return XDocument.Load(reader, options);
-        }
-
-        private static XmlReaderSettings CreateSafeSettings(bool ignoreWhiteSpace = false)
-        {
-            var safeSettings = new XmlReaderSettings
-                {
-                    XmlResolver = null,
-                    DtdProcessing = DtdProcessing.Prohibit,
-                    IgnoreWhitespace = ignoreWhiteSpace
-                };
-
-            return safeSettings;
-        }
-
-        public static XDocument GetOrCreateDocument(XName rootName, string path, IMSBuildProjectSystem msBuildNuGetProjectSystem)
-        {
-            if (File.Exists(Path.Combine(msBuildNuGetProjectSystem.ProjectFullPath, path)))
-            {
-                try
-                {
-                    return GetDocument(msBuildNuGetProjectSystem.ProjectFullPath, path);
-                }
-                catch (FileNotFoundException)
-                {
-                    return CreateDocument(rootName, path, msBuildNuGetProjectSystem);
-                }
-            }
-            return CreateDocument(rootName, path, msBuildNuGetProjectSystem);
-        }
-
         public static XDocument GetOrCreateDocument(XName rootName, string root, string path, INuGetProjectContext nuGetProjectContext)
         {
             if (File.Exists(Path.Combine(root, path)))
             {
                 try
                 {
-                    return GetDocument(root, path);
-                }
-                catch (FileNotFoundException)
-                {
-                    return CreateDocument(rootName, root, path, nuGetProjectContext);
+                    return Shared.XmlUtility.Load(Path.Combine(root, path), LoadOptions.PreserveWhitespace);
```

Comment:
import the namespace instead.

Reply:
I retained [`src/NuGet.Core/NuGet.PackageManagement/Utility/XmlUtility.cs`](https://github.com/NuGet/NuGet.Client/blob/dev-kartheekp-ms-shared-xmlutility/src/NuGet.Core/NuGet.PackageManagement/Utility/XmlUtility.cs) to support backward compatibility as per feedback [above](https://github.com/NuGet/NuGet.Client/pull/3438#discussion_r439837661). Since the class name `XmlUtility` is same in `NuGet.PackageManagement` and `NuGet.Shared` namespaces, adding a import statement or alias name did not help and compilation failed. Hence I added this fully qualified name.

We can rename the new class [`build/Shared/XmlUtility.cs`](https://github.com/NuGet/NuGet.Client/blob/dev-kartheekp-ms-shared-xmlutility/build/Shared/XmlUtility.cs) to `XmlHelper` to avoid the naming conflicts. Please let me know.

Reply:
I don't want you to make too many changes. XmlUtility as a name is fine.

2 options:

* You can consider using a static import for XmlUtility. 
* Do nothing, I'm ok with that too.

Reply:
Considered option 1. Used static import to in the latest commit. All the CI checks have now passed. I will go ahead and merge these changes to `dev` tomorrow.